pilot/pkg/security/authn/utils/utils.go

```diff
@@ -42,64 +158,59 @@ func BuildInboundFilterChain(mTLSMode model.MutualTLSMode, sdsUdsPath string, no
 	}
 
 	meta := node.Metadata
-	var alpnIstioMatch *listener.FilterChainMatch
 	var ctx *tls.DownstreamTlsContext
-	if features.EnableTCPMetadataExchange &&
-		(listenerProtocol == networking.ListenerProtocolTCP || listenerProtocol == networking.ListenerProtocolAuto) {
-		alpnIstioMatch = &listener.FilterChainMatch{
-			ApplicationProtocols: util.ALPNInMeshWithMxc,
-		}
-		ctx = &tls.DownstreamTlsContext{
-			CommonTlsContext: &tls.CommonTlsContext{
-				// For TCP with mTLS, we advertise "istio-peer-exchange" from client and
-				// expect the same from server. This  is so that secure metadata exchange
-				// transfer can take place between sidecars for TCP with mTLS.
-				AlpnProtocols: util.ALPNDownstream,
-			},
-			RequireClientCertificate: protovalue.BoolTrue,
-		}
-	} else {
-		alpnIstioMatch = &listener.FilterChainMatch{
-			ApplicationProtocols: util.ALPNInMesh,
-		}
-		ctx = &tls.DownstreamTlsContext{
-			CommonTlsContext: &tls.CommonTlsContext{
-				// Note that in the PERMISSIVE mode, we match filter chain on "istio" ALPN,
-				// which is used to differentiate between service mesh and legacy traffic.
-				//
-				// Client sidecar outbound cluster's TLSContext.ALPN must include "istio".
-				//
-				// Server sidecar filter chain's FilterChainMatch.ApplicationProtocols must
-				// include "istio" for the secure traffic, but its TLSContext.ALPN must not
-				// include "istio", which would interfere with negotiation of the underlying
-				// protocol, e.g. HTTP/2.
-				AlpnProtocols: util.ALPNHttp,
-			},
-			RequireClientCertificate: protovalue.BoolTrue,
-		}
+	// TLS context to use, if we are going to use it (in permissive/strict)
+	// NOTE: We do not need to set ALPN protocols in the sidecar's downstream TLS context
+	// because the client envoys do not support any form of protocol negotiation anyway
+	// plus we know fully well who we are going to talk to and the protocol used by them
+	// on a given port. If ALPN is missing, envoy on server side will simply accept all.
+	// Irrespective of ALPN negotiation, we use filter chain matches to distinguish the
+	// type of traffic and activate the appropriate decoders (http/tcp).
+	ctx = &tls.DownstreamTlsContext{
```

Comment:
I am not sure if this will work for TCP metadata exchange because the plugin expects that "istio-peer-exchange" is the negotiated ALPN protocol